jenkins/docker-ecr/docker-build-ecr.jenkinsfile

```diff
@@ -0,0 +1,103 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    options {
+        timeout(time: 3, unit: 'HOURS')
+    }
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+            args '-u root -v /var/run/docker.sock:/var/run/docker.sock'
+        }
+    }
+    parameters {
+        string(
+                defaultValue: 'https://github.com/opensearch-project/opensearch-build',
+                name: 'DOCKER_BUILD_GIT_REPOSITORY',
+                description: 'The git repository name that contains dockerfiles and the docker build script.',
+                trim: true
+        )
+        string(
+                defaultValue: 'main',
+                name: 'DOCKER_BUILD_GIT_REPOSITORY_REFERENCE',
+                description: 'The git reference (branch/tag/commit_id) of above repository.',
+                trim: true
+        )
```

Comment:
Are we expecting to build docker images from repositories other than opensearch-build? If not today we can add this later when it is required
trying to see if we can reduce the parameters.

Reply:
I had a conversation with @peterzhuamazon about it. We need the repo etc to specify which repo has the `build.sh` script. For example, this is also used for `data-prepper`, and they have their build script in their repo

Reply:
I do not see `build.sh` script anywhere in this file. It is not generic enough today

Reply:
The`build.sh` would be pulled in from the repo that is listed here. Eg: 

https://github.com/opensearch-project/opensearch-build/blob/main/docker/release/build-image-multi-arch.sh

https://github.com/opensearch-project/data-prepper/blob/main/release/docker/ci/build-image-single-arch.sh

The command would then specify how to run it and parameters for the script

Reply:
Ahh! Its the naming convention issue then. Its a nit maybe fix it later. Moving on